README.md

```diff
@@ -20,4 +20,302 @@ composer require bedita/php-sdk
+    $response = (array)$client->saveObject('documents', $data); // arguments passed are string $type, array $data
+
+    // save an existing document
+    $data = [
+        'id' => 999,
+        'title' => 'My new doc, changed title',
+    ];
+    $response = (array)$client->save('documents', $data); // arguments passed are string $type, array $data
+    // retrocompatibility version with saveObject, deprecated
+    $response = (array)$client->saveObject('documents', $data); // arguments passed are string $type, array $data
+```
+`save` and `saveObject` use internally `patch(string $path, mixed $body, ?array $headers = null)` (when saving an existing object) or `post(string $path, mixed $body, ?array $headers = null)` (when saving a new object).
+
+If you like to use them directly:
+```php
+    // save a new document
+    $data = [
+        'title' => 'My new doc',
+        'status' => 'on',
+    ];
+    $response = (array)$client->post('/documents', $data); // arguments passed are string $path, array $data
+
+    // save an existing document
+    $data = [
+        'title' => 'My new doc',
+        'status' => 'on',
+    ];
+    $response = (array)$client->post('/documents/999', $data); // arguments passed are string $path, array $data
+```
+
+You can add related data using `addRelated(int|string $id, string $type, string $relation, array $data, ?array $headers = null)`.
+
+```php
+    // save a document and add related data, in this example a "see_also" relation between documents and documents is involved
+    $document = $client->save('documents', ['title' => 'My new doc']);
+    $relatedData = [
+        [
+            'id' => 9999, // another doc id
+            'type' => 'documents',
+        ],
+    ];
+    $client->addRelated($document['data']['id'], 'documents', 'see_also', $relatedData); // arguments passed are int|string $id, string $type, string $relation, array $relationData
+```
+
+`replaceRelated(int|string $id, string $type, string $relation, array $data, ?array $headers = null)` is handy to replace relation data.
+
+```php
+    // replace related data, in this example a "see_also" relation between document 8888 and document 9999
+    $relatedData = [
+        [
+            'id' => 9999,
+            'type' => 'documents',
+        ],
+    ];
+    $client->replaceRelated(8888, 'documents', 'see_also', $relatedData); // arguments passed are int|string $id, string $type, string $relation, array $relationData
+```
+
+Note: `addRelated` uses `post`, `replaceRelated` uses `patch`. Both call `/:type/:id/relationships/:relation`
+
+### Delete and restore data
+
+#### Soft delete
+
+Soft delete puts object into the trashcan.
+You can trash an object with `delete(string $path, mixed $body = null, ?array $headers = null)` or `deleteObject(int|string $id, string $type)`.
+
+```php
+    // delete annotation by ID 99999
+    $response = $client->delete('/annotations/99999'); // argument passed is string $path
```

Comment:
maybe `documents` instead of `annotations` 
(annotations are resources, not thrashed but actually removed)